library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableMapImpl.java

```diff
@@ -132,7 +153,11 @@ public V get(Object o)
       return val;
     }
 
-    Slice valSlice = store.getSync(bucket, keyValueSerdeManager.serializeDataKey(key, false));
+    if (timeExtractor != null) {
```

Comment:
Should we put this in a method? like
```java
private long getBucket(K key)
{
  return timeExtractor != null ? timeExtractor.getTime(key) : bucket;
}
```

I see this same pattern in multiple other places so I am suggesting making this a method.